enginetest/server_engine.go

```diff
@@ -183,33 +193,51 @@ func (s *ServerQueryEngine) QueryWithBindings(ctx *sql.Context, query string, pa
 	//  However, Dolt supports, but not go-sql-driver client
 	switch parsed.(type) {
 	case *sqlparser.Load, *sqlparser.Execute, *sqlparser.Prepare:
-		rows, err := s.conn.Query(query)
-		if err != nil {
-			return nil, nil, trimMySQLErrCodePrefix(err)
-		}
-		return convertRowsResult(rows)
+		return s.queryOrExec(nil, parsed, q, []any{})
 	}
 
-	stmt, err := s.conn.Prepare(query)
+	stmt, err := s.conn.Prepare(q)
 	if err != nil {
 		return nil, nil, trimMySQLErrCodePrefix(err)
 	}
 
 	args := prepareBindingArgs(bindings)
 
+	return s.queryOrExec(stmt, parsed, q, args)
+}
+
+// queryOrExec function use `query()` or `exec()` method of go-sql-driver depending on the sql parser plan.
+// If |stmt| is nil, then we use the connection db to query/exec the given query statement because some queries cannot
+// be run as prepared.
+// TODO: for `EXECUTE` and `CALL` statements, it can be either query or exec depending on the statement that prepared or stored procedure holds.
+//
+//	for now, we use `query` to get the row results for these statements. For statements that needs `exec`, there will be no result.
+func (s *ServerQueryEngine) queryOrExec(stmt *gosql.Stmt, parsed sqlparser.Statement, query string, args []any) (sql.Schema, sql.RowIter, error) {
+	var err error
 	switch parsed.(type) {
-	case *sqlparser.Select, *sqlparser.SetOp, *sqlparser.Show, *sqlparser.Set, *sqlparser.Call, *sqlparser.Begin, *sqlparser.Use:
-		rows, err := stmt.Query(args...)
+	// TODO: added `FLUSH` stmt here (should be `exec`) because we don't support `FLUSH BINARY LOGS` or `FLUSH ENGINE LOGS`, so nil schema is returned.
```

Comment:
Very recent change that we parse `FLUSH BINARY LOGS` and `FLUSH ENGINE LOGS` statements, but not supported. This gets `Nothing{}` node, which returns nil schema and empty result. Using `Exec()` on this returns OkResult, but it returns nothing in the enginetest. To match it to the current test, running these statements as `Query()`.